src/polyswarm_api/__init__.py

```diff
@@ -406,6 +408,39 @@ def _fix_result(self, result):
         response['search'] = '{hash_type}={hash}'.format(hash_type=hash_type, hash=to_scan)
         return response
 
+    async def search_query(self, query):
+        """
+        Search by Elasticsearch query using the PS API asynchronously.
+
+        :param query: Elasticsearch query
+        :return: JSON report file
+        """
+        async with self.get_semaphore:
+            async with aiohttp.ClientSession() as session:
+                try:
+                    async with session.get('{query_uri}/{query}'.format(**{'query_uri': self.query_uri,
+                                                                           'query': parse.quote(json.dumps(query))}),
+                                           headers={"Authorization": self.api_key}) as raw_response:
+
+                        try:
+                            response = await raw_response.json()
+                        except Exception:
+                            response = await raw_response.read() if raw_response else 'None'
+                            response = response.decode('utf-8')
+                            if raw_response.status == 404:
+                                return {"search": query, 'result': []}
```

Comment:
Why  http `404` returns an empty result instead of an error?

Reply:
I'm intrigued my this design decision as well, but I decided to not change the original behavior.

Reply:
@supernothing ?